core/common/src/main/java/alluxio/security/authentication/SaslStreamClientDriver.java

```diff
@@ -41,20 +42,24 @@
   private SaslHandshakeClientHandler mSaslHandshakeClientHandler;
   /** Used to wait until authentication is completed. */
   private SettableFuture<Boolean> mAuthenticated;
+  /** Whether the authentication is active. */
+  private AtomicBoolean mAuthenticationActive;
 
   private final long mGrpcAuthTimeoutMs;
 
   /**
    * Creates client driver with given handshake handler.
    *
    * @param handshakeClient client handshake handler
+   * @param authenticationActive boolean reference to receive authentication state changes
```

Comment:
Is this class doing the "receiving" of the state change? It looks like the state is set here?

Reply:
This class knows whether the client channel is authenticated or not. So from client's perspective this class manages the state, end purpose of this parameter is to receive that current state from outside.

Reply:
I see, I thought it meant this class (SaslStreamClientDriver) is receiving the state, but what it means is the caller is receiving the state via this bool. Thanks for the clarification.

However, can we rename to `authenticated` to match the internal variable name?